index.rb

```diff
@@ -19,56 +21,106 @@ def my_select
     result
   end
 
-  def my_all?
-    my_each do |e|
-      return false unless yield(e)
+  def my_all?(*args)
+    if !args[0].nil?
+      my_each { |item| return false unless args[0] == item }
+    elsif block_given?
+      my_each { |item| return false unless yield(item) }
+    elsif args.instance_of?(Class)
+      my_each { |item| return false unless yield(item) != args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return false unless args.match(item) }
+    else
+      my_each { |item| return false unless item }
     end
     true
   end
 
-  def my_any?
-    my_each do |e|
-      return true if yield(e)
+  def my_any?(*args)
+    if !args[0].nil?
+      my_each { |item| return true if args[0] == item }
+    elsif block_given?
+      my_each { |item| return true if item }
+    elsif args.instance_of?(Class)
+      my_each { |item| return true if yield(item) == args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return true if args.match(item) }
+    else
+      my_each { |item| return true if item == args }
```

Comment:
- [ ] The method doesn't pass the test for Classes and boolean cases .e.g 
```ruby
[nil, true, 99].my_any?(Integer)          #returned false instead of true
[nil, true, 99].my_any?                         #returned false instead of true
```